util/predicates/cluster_predicates_test.go

```diff
@@ -1,79 +1,148 @@
 			Expect(result).To(BeTrue())
 		})
 	})
 	Context("when CAPI cluster has a random annotation", func() {
 		It("should return true", func() {
-			capiCluster := &clusterv1.Cluster{
-				ObjectMeta: metav1.ObjectMeta{
-					Annotations: map[string]string{"some-random-annoation": "true"},
-				},
-			}
-			capiCluster = &clusterv1.Cluster{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:        "test-cluster",
-					Namespace:   "test-ns",
-					Annotations: map[string]string{},
-				},
+			capiCluster.Annotations = map[string]string{
+				"some-other-annoations": "true",
 			}
 			result := ClusterWithoutImportedAnnotation(logger).UpdateFunc(event.UpdateEvent{ObjectNew: capiCluster})
 			Expect(result).To(BeTrue())
 		})
 	})
 })
 
-func TestClusterPredicates(t *testing.T) {
-	RegisterFailHandler(Fail)
-	RunSpecs(t, "ClusterPredicates Suite")
-}
+var _ = Describe("ClusterWithReadyControlPlane", func() {
+	var (
+		logger      logr.Logger
+		capiCluster *clusterv1.Cluster
+	)
+
+	BeforeEach(func() {
+		// Initialize the logger
+		logger = logr.Discard()
+
+		capiCluster = &clusterv1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-cluster",
+				Namespace: "test-ns",
+			},
+		}
+	})
+
+	It("should return true when cluster has ready control plane", func() {
+		capiCluster.Status.ControlPlaneReady = true
+		result := ClusterWithReadyControlPlane(logger).UpdateFunc(event.UpdateEvent{ObjectNew: capiCluster})
+		Expect(result).To(BeTrue())
+	})
+
+	It("should return false when cluster has no ready control plane", func() {
+		result := ClusterWithReadyControlPlane(logger).UpdateFunc(event.UpdateEvent{ObjectNew: capiCluster})
+		Expect(result).To(BeFalse())
+	})
+})
+
+var _ = Describe("ClusterOrNamespaceWithImportLabel", func() {
+	var (
+		logger      logr.Logger
+		capiCluster *clusterv1.Cluster
+		namespace   *corev1.Namespace
+	)
+
+	BeforeEach(func() {
+		// Initialize the logger
+		logger = logr.Discard()
+
+		capiCluster = &clusterv1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "test-cluster",
+			},
+		}
+
+		namespace = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					importLabel: "true",
+				},
+			},
+		}
+	})
+
+	It("should return true when cluster has import label", func() {
+		capiCluster.Labels = map[string]string{
+			importLabel: "true",
+		}
+		result := ClusterOrNamespaceWithImportLabel(ctx, logger, cl, importLabel).UpdateFunc(event.UpdateEvent{ObjectNew: capiCluster})
+		Expect(result).To(BeTrue())
+	})
+
+	It("should return true when namespace has import label", func() {
+		namespace.Name = "test-ns-1"
+		Expect(cl.Create(ctx, namespace)).To(Succeed())
+
+		capiCluster.Namespace = namespace.Name
+		result := ClusterOrNamespaceWithImportLabel(ctx, logger, cl, importLabel).UpdateFunc(event.UpdateEvent{ObjectNew: capiCluster})
+		Expect(result).To(BeTrue())
+	})
+
+	It("should return true if client fails to get namespace", func() {
```

Comment:
Are you thinking of transient errors here?

Reply:
yes